common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2Strategy.java

```diff
@@ -230,6 +245,101 @@ protected final void setAuthorizationEndpoint(final String authorizationEndpoint
         mAuthorizationEndpoint = authorizationEndpoint;
     }
 
+    public AuthorizationResult getDeviceCode(MicrosoftStsAuthorizationRequest authorizationRequest, @Nullable String authorityUrl) throws IOException {
+        final String methodName = ":getDeviceCode";
+
+        // Set up connection
+        String urlBody;
+
+        if (authorityUrl == null) {
+            // Fetch the tenant from the authority
+            String tenant = authorizationRequest.getAuthority().getPath().split("/")[1];
+
+            // Set up Device Code Flow authorization url
+            urlBody = "https://login.microsoftonline.com/" + tenant + "/oauth2/v2.0/devicecode";
```

Comment:
This is Microsoft specific, this should move _down_ the object inheritance hierarchy and into `MicrosoftStsOAuth2Strategy`

Reply:
This URL should be sourced from the `MicrosoftStsOAuth2Configuration` -- something like `getDeviceCodeEndpoint()`

Reply:
I would expect that the URL for this endpoint should be available at:
`https://login.microsoftonline.com/common/.well-known/openid-configuration`
via `device_authorization_endpoint` but it's not....

We should reach out to server team to determine why this is missing; I can help facilitate that communication

/cc @shoatman

Reply:
Here is the relevant OID docs:
https://tools.ietf.org/id/draft-ietf-oauth-device-flow-08.html#Discovery

Reply:
Heard back from the service team, this should be available around the first week of September. For now, we will hardcode and make the necessary change to resolve this value dynamically later

Reply:
Created a tracking work item here:
https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1096018